tests/mir-opt/sroa/structs.rs

```diff
@@ -13,28 +12,65 @@ impl Drop for Tag {
     fn drop(&mut self) {}
 }
 
+/// Check that SROA excludes structs with a `Drop` implementation.
 pub fn dropping() {
+    // CHECK-LABEL: fn dropping(
+
+    // CHECK: [[aggregate:_[0-9]+]]: S;
+
+    // CHECK: bb0: {
+    // CHECK: [[aggregate]] = S
     S(Tag(0), Tag(1), Tag(2)).1;
 }
 
+/// Check that SROA excludes enums.
 pub fn enums(a: usize) -> usize {
+    // CHECK-LABEL: fn enums(
+
+    // CHECK: [[enum:_[0-9]+]]: std::option::Option<usize>;
+
+    // CHECK: bb0: {
+    // CHECK: [[enum]] = Option::<usize>::Some
     if let Some(a) = Some(a) { a } else { 0 }
 }
 
+/// Check that SROA destructures `U`.
 pub fn structs(a: f32) -> f32 {
+    // CHECK-LABEL: fn structs(
+    // CHECK-SAME: [[a:_[0-9]+]]: f32
     struct U {
         _foo: usize,
         a: f32,
     }
-
+    // CHECK: [[ret:_0]]: f32;
+    // CHECK-DAG: [[struct:_[0-9]+]]: structs::U;
+    // CHECK-DAG: [[foo:_[0-9]+]]: usize;
+    // CHECK-DAG: [[a_tmp:_[0-9]+]]: f32;
+    // CHECK-NOT: Stop reordering
+    // CHECK-DAG: [[a_ret:_[0-9]+]]: f32;
```

Comment:
Could you use simple `CHECK`s? The order should be deterministic. (This comment applies everywhere you use CHECK-DAG).